File: 1/go/solution.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func solution() {
	s1()
	s2()
}

func s1() {
	in := string(input)
	numbers := strings.Split(in, "\n")

	lastValue, _ := strconv.ParseInt(numbers[0], 10, 32)
	count := 0

	for _, numStr := range numbers[1:] {
		num, _ := strconv.ParseInt(numStr, 10, 32)
		if num > lastValue {
			count++
		}
		lastValue = num
	}

	print(fmt.Sprintf("__S1__ = %d\n", count))
}

func s2() {
	in := string(input)
	numbers := strings.Split(in, "\n")

	var values [3]int64
	values[0], _ = strconv.ParseInt(numbers[0], 10, 32)
	values[1], _ = strconv.ParseInt(numbers[1], 10, 32)
	values[2], _ = strconv.ParseInt(numbers[2], 10, 32)

	lastSum := values[0] + values[1] + values[2]
	count := 0

	for _, numStr := range numbers[3:] {
		num, _ := strconv.ParseInt(numStr, 10, 32)
		values[0] = values[1]
		values[1] = values[2]
		values[2] = num
		sum := values[0] + values[1] + values[2]

		if sum > lastSum {
			count++
		}
		lastSum = sum
	}

	print(fmt.Sprintf("__S2__ = %d\n", count))
}

File: 1/python/solution.py
def solution(input: str):
    s1(input)
    s2(input)

def s1(input: str):
    lines = input.splitlines()
    numbers = [int(x) for x in lines]
    lastValue = -1
    count = 0

    for num in numbers:
        if lastValue != -1 and num > lastValue:
            count = count + 1
        lastValue = num
    print("__S1__ = %d" % count)

def s2(input: str):
    lines = input.splitlines()
    numbers = [int(x) for x in lines]
    lastSum = numbers[0] + numbers[1] + numbers[2]
    count = 0

    for i in range(2, len(numbers)):
        sum = numbers[i-2] + numbers[i-1] + numbers[i]
        if sum > lastSum:
            count = count + 1
        lastSum = sum
    print("__S2__ = %d" % count)